src/plugins/dashboard/public/dashboard_app/top_nav/open_dashboard_panel_selection_flyout.tsx

```diff
@@ -0,0 +1,237 @@
+  getPanels,
+  flyoutPanelPaddingSize = 'l',
+}: OpenDashboardPanelSelectionFlyoutArgs) {
+  const {
+    overlays,
+    analytics,
+    settings: { i18n, theme },
+  } = pluginServices.getServices();
+  // eslint-disable-next-line prefer-const
+  let flyoutRef: ReturnType<DashboardServices['overlays']['openFlyout']>;
+
+  const mount = toMountPoint(
+    React.createElement(function () {
+      const closeFlyout = () => flyoutRef.close();
+      return (
+        <DashboardPanelSelectionListFlyout
+          close={closeFlyout}
+          {...{ paddingSize: flyoutPanelPaddingSize, getPanels }}
+        />
+      );
+    }),
+    { analytics, theme, i18n }
+  );
+
+  flyoutRef = overlays.openFlyout(mount, {
+    size: 'm',
+    maxWidth: 500,
+    paddingSize: flyoutPanelPaddingSize,
+    'aria-labelledby': 'addPanelsFlyout',
+    'data-test-subj': 'dashboardPanelSelectionFlyout',
+  });
+
+  return flyoutRef;
+}
+
+export const DashboardPanelSelectionListFlyout: React.FC<Props> = ({
+  close,
+  getPanels,
+  paddingSize,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const panels = useRef(getPanels(close));
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [panelsSearchResult, setPanelsSearchResult] = useState(panels.current);
+
+  useEffect(() => {
+    if (!searchTerm) {
+      return setPanelsSearchResult(panels.current);
+    }
+
+    const q = searchTerm.toLowerCase();
+
+    setPanelsSearchResult(
+      orderBy(
+        panels.current.map((panel) => {
+          const groupSearchMatch = panel.title.toLowerCase().includes(q);
+
+          const [groupSearchMatchAgg, items] = panel.items.reduce(
+            (acc, cur) => {
+              const searchMatch = cur.name.toLowerCase().includes(q);
+
+              acc[0] = acc[0] || searchMatch;
+              acc[1].push({
+                ...cur,
+                isDisabled: !(groupSearchMatch || searchMatch),
+              });
+
+              return acc;
+            },
+            [groupSearchMatch, [] as PanelSelectionMenuItem[]]
+          );
+
+          return {
+            ...panel,
+            isDisabled: !groupSearchMatchAgg,
+            items,
+          };
+        }),
+        ['isDisabled']
+      )
+    );
+  }, [searchTerm]);
+
+  return (
+    <>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="m">
+          <h1 id="addPanelsFlyout">
+            <FormattedMessage
+              id="dashboard.solutionToolbar.addPanelFlyout.headingText"
+              defaultMessage="Add panel"
+            />
+          </h1>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiFlexGroup direction="column" responsive={false}>
+          <EuiFlexItem
+            grow={false}
+            css={{
+              position: 'sticky',
+              top: euiTheme.size[paddingSize],
+              zIndex: 1,
+              boxShadow: `0 -${euiTheme.size[paddingSize]} 0 4px ${euiTheme.colors.emptyShade}`,
```

Comment:
use box-shadow with a negative vertical shadow that equals the flyout's body padding to hide the panel list that get displayed on scrolled, especially that the input field remains sticky on scroll.